backend/src/test/java/ba/unsa/etf/si/bbqms/api/GroupControllerTest.java

```diff
@@ -0,0 +1,350 @@
+package ba.unsa.etf.si.bbqms.api;
+
+import ba.unsa.etf.si.bbqms.ws.controllers.GroupController;
+import ba.unsa.etf.si.bbqms.ws.models.BranchGroupCreateDto;
+import ba.unsa.etf.si.bbqms.ws.models.BranchGroupUpdateDto;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.*;
+import static org.hamcrest.Matchers.greaterThan;
+
+public class GroupControllerTest extends BaseSetup {
+    private int branchGroupId;
+    private final String tenantCode = "DFLT";
+    private final String invalidTenantCode = ".";
+
+    @BeforeEach
+    public void createBranchGroup_ValidNameBranchIdsAndServiceIds_ReturnsOkResponse() {
+        final String name = "Dummy Branch Group";
+        final Set<Long> branchIds = Set.of(2L);
+        final Set<Long> serviceIds = Set.of(1L);
```

Comment:
If I don't have branches and/or services with these ID's in my database (and you can't expect me to have them), this will fail. We should make sure the test has everything inside it needed to pass (including creating (and later deleting) the branches and services needed for the group.

You should `@BeforeAll` create the needed branches and services (and whatever else may be needed), and then in `@AfterAll` free those resources (delete them from the database).